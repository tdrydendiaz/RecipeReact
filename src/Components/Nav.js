import React, { Component } from 'react';

import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import {
  Collapse,
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  NavItem,
  NavLink,
  UncontrolledDropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem
} from 'reactstrap';

import Routers from './Routers'

export default class IndexNav extends Component {
  constructor(props) {
    super(props);

    this.toggle = this.toggle.bind(this);
    this.state = {
      isOpen: false
    };
  }
  toggle() {
    this.setState({
      isOpen: !this.state.isOpen
    });
  }
  render() {
    return (
      <div>
        <Navbar color="light" light expand="md">
          <NavbarBrand href="/">Recipes ^_^</NavbarBrand>
          <NavbarToggler onClick={this.toggle} />
          <Collapse isOpen={this.state.isOpen} navbar>
            <Nav className="ml-auto" navbar>
              <NavItem>
                <NavLink href="/">Home</NavLink>
              </NavItem>
              <NavItem>
                <NavLink href="/MoreDetail">MoreDetail</NavLink>
              </NavItem>

              <NavItem>
                <NavLink href="/AddRecipe">New Recipe</NavLink>
              </NavItem>
              <NavItem>
                <NavLink href="https://toolband.com/">ClickForASurprise!</NavLink>
              </NavItem>

            </Nav>
          </Collapse>
        </Navbar>
        <Routers />
      </div>
    );
  }
}