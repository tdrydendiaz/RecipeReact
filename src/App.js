import React from 'react';
import logo from './logo.svg';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";

import './App.css';
import Nav from './Components/Nav';
import axios from "axios";
import RecipeThread from './Components/RecipeThread'
import MoreDetail from './Components/MoreDetail'
import AddRecipe from './Components/AddRecipe'

export default class App extends React.Component {

  constructor() {
    super();
    this.state = {
      data: []
    }
  };

  componentDidMount() {
    this.onLoad();
  };

  onLoad = () => {
    axios
      .get("http://localhost:5000/recipe/all")
      .then(response => {
        this.setState({
          data: response.data
        });
        console.log(this.state.data)
      });
  }

  render() {
    return (
      <div className="App">
        <Router>

          <Nav />

          <Route exact path="/" render={() => <RecipeThread getAll={this.onLoad} data={this.state.data} />} />

          <Route exact path="/AddRecipe" render={() => <AddRecipe getAll={this.onLoad} data={this.state.data} />} />

          <Route path="/MoreDetail" component={MoreDetail} />


          {this.state.data.map((item) => (

            <Route path={"/" + item.name} render={() => <MoreDetail getAll={this.onLoad}
              name={item.name}
              description={item.description}
              ingredients={item.ingredients}

              id={item._id} />} />

          ))}

        </Router>

      </div>
    );
  }
}
