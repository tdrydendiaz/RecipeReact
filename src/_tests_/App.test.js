import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import App from '../App';
import Nav from '../Components/Nav'


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders Nav comp', () => {
  const wrapper = shallow(<App />);
  const content = <Nav />;
  expect(wrapper.contains(content)).toEqual(true);
});

it("render a prop", () => {
    const wrapper = shallow(<App name="A Name" />);
    expect(wrapper.instance().props.name).toEqual("A Name");
    expect(wrapper).toMatchSnapshot();
  });
