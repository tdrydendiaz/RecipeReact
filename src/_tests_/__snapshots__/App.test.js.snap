// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a prop 1`] = `ShallowWrapper {}`;
